src/Dev/SapphireTest.php

```diff
@@ -253,31 +281,10 @@ protected function setUp()
 
         $fixtureFiles = $this->getFixturePaths();
 
-        // Set up fixture
         if ($this->shouldSetupDatabaseForCurrentTest($fixtureFiles)) {
-            if (!static::$tempDB->isUsed()) {
-                static::$tempDB->build();
-            }
-
-            DataObject::singleton()->flushCache();
-
-            static::$tempDB->clearAllData();
-
-            foreach ($this->requireDefaultRecordsFrom as $className) {
-                $instance = singleton($className);
-                if (method_exists($instance, 'requireDefaultRecords')) {
-                    $instance->requireDefaultRecords();
-                }
-                if (method_exists($instance, 'augmentDefaultRecords')) {
-                    $instance->augmentDefaultRecords();
-                }
-            }
-
-            foreach ($fixtureFiles as $fixtureFilePath) {
-                $fixture = YamlFixture::create($fixtureFilePath);
-                $fixture->writeInto($this->getFixtureFactory());
-            }
-
+            /** @var FixtureTestState $fixtureState */
+            $fixtureState = static::$state->getStateByName('fixtures');
+            $this->setFixtureFactory($fixtureState->getFixtureFactory(static::class));
```

Comment:
Is this needed still? Doesn't `this->setFixtureFactory()` just send the factory back to the teststate?

Reply:
we need to set `$this->fixutureFactory` (property on `SapphireTest`) because some tests access it directly rather than through the helper methods or `getFixtureFactory` :(